approval_polls/templates/approval_polls/edit.html

```diff
@@ -16,111 +16,117 @@
-      <div class='form-group'>
-        <div class='input-group' id='div-choice{{ choice.pk }}'>
-          <input class='form-control' type='text' maxlength='200' name='choice{{ choice.pk }}' placeholder='Choice {{choice.pk}}' value='{{choice.choice_text}}'>
-          <span class='input-group-addon'>
-            <a href='#' class='add-link' id='link-choice{{ choice.pk }}' title='Add link' data-toggle='tooltip' data-placement='bottom'>
-              <span class='glyphicon glyphicon-link'></span>
-            </a>
-          </span>
-          <span class='input-group-addon'>
-            <a href='#' class='remove-choice' id='remove-choice{{ choice.pk }}' title='Remove Choice' >
-              <span class='glyphicon glyphicon-remove'></span>
-            </a>
-          </span>
-          <input type='hidden' id='linkurl-choice{{ choice.pk }}' name='linkurl-choice{{ choice.pk }}' value=''>
+      <form action="{% url 'approval_polls:edit' poll_id=poll.id %}" method='post'>
+        <legend> Edit: {{poll.question}}</legend>
+        <p>
+          {% if not can_edit_poll %}
+            You cannot edit the questions and choices as this poll has got ballots on it!
+          {% endif %}
+        </p>
+        {% csrf_token %}
+        <div class='form-group {% if question_error %} has-error {% endif %}'>
+          {% if question_error %}
+            <label class='control-label' for='question'>
+                {{ question_error }}
+            </label>
+          {% endif %}
+          <input type='text' class='form-control' maxlength='200' value='{{ poll.question }}'
+                 name='question' size='28' placeholder='Poll Question' {% if not can_edit_poll %}disabled{% endif %}>
         </div>
-      </div>
-      {% endfor %}
 
-      <p>
-        <button type='button' class='btn btn-primary btn-sm' id='add-choice'
-              title='Add choice' data-toggle='tooltip' data-placement='bottom'>
-          <i class='glyphicon glyphicon-plus'></i>
-        </button>
-      </p>
--->      
-      <!-- Add the collapsible poll settings panel here -->
-      <div class='panel-group' id='accordion' role='tablist' aria-multiselectable='true'>
-        <div class='panel panel-default'>
-          <div class='panel-heading' role='tab' id='headingOne'>
-            <h4 class='panel-title'>
-              <a role='button' data-toggle='collapse' data-parent='#accordion' href='#collapsePollSettings' aria-expanded='true' aria-controls='collapsePollSettings'>
-               Poll Settings
-              </a>
-            </h4>
+        <input id='choices-count' type='hidden' value='{{ choices.count }}'>
+        {% for choice in choices %}
+        <div class='form-group'>
+          <div class='input-group' id='div-choice{{ choice.pk }}'>
+            <input class='form-control' type='text' maxlength='200' name='choice{{ choice.pk }}' placeholder='Choice' value='{{choice.choice_text}}' {% if not can_edit_poll %}disabled{% endif %}>
```

Comment:
Choice.pk is based on the database id of the choice. It might be 5. If it's 5 and a user edits the poll and adds another choice there will be two `choice5`s this will cause problems. The fix to this probably happens in create.js, you need some way to start the ids higher than the highest existing id.